rts/lra/coordinator/src/test/java/io/narayana/lra/coordinator/domain/model/LRATest.java

```diff
@@ -0,0 +1,897 @@
+
+        try {
+            Optional<java.nio.file.Path> lra = Files.list(new File(lraDir.toString()).toPath()).findFirst();
+
+            return lra.map(path -> path.getFileName().toString()).orElse(null);
+        } catch (IOException e) {
+            return null;
+        }
+    }
+
+    LRAStatus getStatus(URI lra) {
+        try {
+            return lraClient.getStatus(lra);
+        } catch (NotFoundException ignore) {
+            return null;
+        }
+    }
+    private boolean isFinished(URI lra) {
+        LRAStatus status = getStatus(lra);
+
+        return status == null
+                || status == LRAStatus.Closed || status == LRAStatus.Cancelled
+                || status == LRAStatus.FailedToClose || status == LRAStatus.FailedToCancel;
+    }
+    private void assertStatus(String message, URI lraId, LRAStatus ... expectedValues) {
+            LRAStatus status = getStatus(lraId);
+
+            assertTrue(message + ": LRA status: " + status,
+                    Arrays.stream(expectedValues).anyMatch(s -> s == status));
+    }
+    private void assertStatus(String lraId, LRAStatus expected, boolean nullValid) {
+        try {
+            LRAStatus status = getStatus(new URI(lraId));
+
+            assertTrue("unexpected null LRA status", status != null || nullValid);
+
+            assertTrue("Expected status " + expected + " but state was " + status,
+                    status == null || status == expected);
+        } catch (URISyntaxException e) {
+            fail(String.format("%s: %s", testName.getMethodName(), e.getMessage()));
+        }
+    }
+    private String getCompensatorLinkHeader() {
+        String prefix = TestPortProvider.generateURL("/base/test");
+        // http://localhost:8081/base/test/forget
+        return String.join(",",
+                makeLink(prefix, "forget"),
+                makeLink(prefix, "after"),
+                makeLink(prefix, "complete"),
+                makeLink(prefix, "compensate")
+        );
+    }
+    private static String makeLink(String uriPrefix, String key) {
+        return Link.fromUri(String.format("%s/%s", uriPrefix, key))
+                .title(key + " URI")
+                .rel(key)
+                .type(MediaType.TEXT_PLAIN)
+                .build().toString();
+    }
+    private void clearObjectStore() {
+        final String objectStorePath = arjPropertyManager.getObjectStoreEnvironmentBean().getObjectStoreDir();
+        final File objectStoreDirectory = new File(objectStorePath);
+
+        clearDirectory(objectStoreDirectory);
+    }
+
+    private void clearDirectory(final File directory) {
+        final File[] files = directory.listFiles();
+
+        if (files != null) {
+            for (final File file : Objects.requireNonNull(directory.listFiles())) {
+                if (file.isDirectory()) {
+                    clearDirectory(file);
```

Comment:
is recursion required when the directory itself is going to deleted at the next statement?

Reply:
Yes. You should try pasting the code into a simple java main and debug it.